### ISSUE TITLE
LG TV - issues with LIBCEC 4 (Source switch when TV switched on/auto play paused video)

### ISSUE THREAD

**Issue by ghost** [Pulse-Eight/libcec#344]
I have two issue with my LG LM640S television when using it with latest KODI 18 testbuilds. I tried and the same issue happens on the stable Libreelec and I can confirm that everything was fine with libcec 3.

My config:
TV: LG LM640S
HDMI 1: RPI2 with KODI with testbuilds from http://forum.kodi.tv/showthread.php?tid=298461
CEC adapter settings: [https://www.dropbox.com/s/gfbn4f6put60ej7/IMG_5327.JPG?dl=0](CEC adapter config in KODI)

## ISSUE 1
- RPI is always switched on from a wall adapter
When we are turning on the TV, it switch to the RPI as an active source. It does not matter if the last input was tv or not.
Same happens if tv is turned on and I restart the RPI or plug out and then plugin in RPI.

LOG: [https://www.dropbox.com/s/q9wg9yzxwgw4plh/kodi.log2?dl=0](https://www.dropbox.com/s/q9wg9yzxwgw4plh/kodi.log2?dl=0)
The log snapshot were made just after the tv auto switched the source to RPI, so the CEC log is in the last minute.

## ISSUE 2
When I pause a video with tv remote, the video starts to play after 10-20 seconds. When logging disabled, this issue comes almost every time. When logging enabled, it happens 1 in 5 times. Very annoying :)

Part of the log, where I pause the video on CEC and when it start to play: [https://pastebin.com/G8yYxNK8](https://pastebin.com/G8yYxNK8)
Full LOG: [https://www.dropbox.com/s/gif5pt15fks1hy3/kodi.log3?dl=0](https://www.dropbox.com/s/gif5pt15fks1hy3/kodi.log3?dl=0)

**Comment by ashtonaut** [comment]
Your Issue #1 sounds like it is this:

https://github.com/Pulse-Eight/libcec/issues/307

This is currently unresolved.

**Comment by rrolandd** [comment]
Yes, that issue sounds similar. Maybe my log files help to the developer to figure out a solution.

**Comment by opdenkamp** [comment]
can you check whether this solve your second issue https://github.com/Pulse-Eight/libcec/commit/1a8dc6c96a1c74d2820fea1a39f4dc8057905102

**Comment by opdenkamp** [comment]
Both issues here were fixed a long time ago and the fixes are in current master, so closing this out.

### Issue 1 — TV switches input to the RPi when powered on

This was the same problem as #307. On power-on, the LG sends an SL "request reconnect" vendor command (`0x0b`), and libCEC was treating that as a full power-on and marking itself the active source, so the TV switched to the RPi regardless of the previous input. Fixed in `5bbff702` ("Fix LG TV always changing input when turned on"): `0x0b` now only reports our power status and no longer switches the source. That's exactly the mechanism you found by reverting #256.

### Issue 2 — paused video resumes on its own after 10–20s

The LG sends vendor command `0x0b` while playing, and libCEC re-activated the source in response, which resumed playback. Fixed in `1a8dc6c9` ("LG - don't activate the source when receiving vendor command 0xB"): the source is only re-activated when it wasn't already active or when the TV explicitly powered us on (`0x03`), so a `0x0b` no longer kicks a paused video back into play.

Both are in every release since libCEC 4.x. Please give a current master (or 7.0.0) build a try — reopen if an LG firmware update has brought either behaviour back.
